src/Workspaces/VisualBasic/Portable/CodeStyle/VisualBasicEditorConfigOptionsGenerator.vb

```diff
@@ -0,0 +1,30 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Collections.Immutable
+Imports System.Composition
+Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.ExternalAccess.EditorConfig
+Imports Microsoft.CodeAnalysis.Host.Mef
+Imports Microsoft.CodeAnalysis.Options
+Imports Microsoft.CodeAnalysis.PooledObjects
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.CodeStyle
+    <EditorConfigGenerator(LanguageNames.VisualBasic), [Shared]>
+    Friend Class VisualBasicEditorConfigOptionsGenerator
+        Implements IEditorConfigOptionsCollection
+
+        <ImportingConstructor>
+        <Obsolete(MefConstruction.ImportingConstructorMessage, True)>
+        Public Sub New()
+        End Sub
+
+        Public Function GetEditorConfigOptions() As ImmutableArray(Of (String, ImmutableArray(Of IOption2))) Implements IEditorConfigOptionsCollection.GetEditorConfigOptions
+            Dim builder = ArrayBuilder(Of (String, ImmutableArray(Of IOption2))).GetInstance()
+            builder.AddRange(EditorConfigFileGenerator.GetLanguageAgnosticEditorConfigOptions())
+            builder.Add((VBWorkspaceResources.VB_Coding_Conventions, VisualBasicCodeStyleOptions.AllOptions.As(Of IOption2)))
+            Return builder.ToImmutableAndFree()
```

Comment:
this logic appears to be the same, except for the resource strings.  can you move into abstract base class?